### DISCUSSION TITLE
Multivariate timeseries forecasting strategy

### DISCUSSION THREAD

**Discussion by henjoe** [autogluon/autogluon#4364] [Q&A]
Hi as far as I know (correct me if I am wrong), mutivariate timeseries forecasting is not yet supported by AutogluonTS, I wonder how are Autogluon users do the timeseries forecasting for mutivariate? I believe someone already done this before, it would be much appreciated if you can share some strategy or concept you did to do the mutivariate TS in Autogluon.

Here's what I am thinking, but to be honest I am not sure if this will work, or the concept is against the Autogluon application.

For multivariate, for example I have multiple features and  one target: 
F1,F2,F3 and T1. 
What I will do is: 

1. Create a univariate TS model for each features( F1,F2,F3)
2. Once I have that, so it means I can predict future values for each features right? 
3. Now I will also train a basic regression problem based on my F1,F2,F3 values to predict my T1.
4. The output of my TS model predicted value for F1,F2 and F3 will be feed in (as input) on the model created on number 3. This will output multiple predictions of T1. 
5. Then I will use that prediction as also the future values of T1. 

Is this setup / strategy is okay to do to predictiom for mutivariate  timeseries? 
I am happy to know other techniques or strategy you've done before. 

Regards,

**Comment by shchur** [comment]
Hi @henjoe, this strategy looks reasonable but I cannot make any guarantees about the accuracy / performance of the resulting model. Up to this point, our focus in AutoGluon was on univariate /panel time series forecasting, so we optimized all components of the system for that purpose. While one can convert multivariate problems into univariate ones and use AutoGluon, it might be that some "multivariate-first" approaches (e.g., those available in [`darts`](https://github.com/unit8co/darts?tab=readme-ov-file#forecasting-models)) will produce more accurate forecasts.

**Reply by henjoe** [reply]
Thanks for replying. This is well noted 🤙
